recipe/meta.yaml

```diff
@@ -62,21 +65,16 @@ outputs:
         - nccl             # [linux and cuda_compiler != "None"]
         - librmm ={{ rapids_version }}  # [linux and cuda_compiler != "None"]
       run:
-        - librmm ={{ rapids_version }}  # [linux and cuda_compiler != "None"]
-        - __cuda  # [cuda_compiler != "None"]
-
-  - name: _py-xgboost-mutex
-    version: 2.0
-    build:
-      string: rapidsai_cpu_0  # [cuda_compiler == "None"]
-      string: rapidsai_gpu_0  # [cuda_compiler != "None"]
+        - cuda-version >=11.2,<12  # [(cuda_compiler_version or "").startswith("11")]
+      run_constrained:
+        - librmm ={{ rapids_version }}               # [linux and cuda_compiler != "None"]
```

Comment:
Why is librmm in run_constrained?

Reply:
Sure 2 reasons:

1. While `librmm` is used to build the package, there isn't a need to have it in `run` after (it's baked into the library). Though as `librmm` could change ABI between versions, it's included in `run_constrained` for alignment with other packages & environments  that depend on `librmm`
2. CUDA dependencies are moved to `run_constrained` so these are not pulled in by default (particularly when considering CPU use cases)

Reply:
Why would ABI changes in librmm matter if we only need it at build time? Are there places where the librmm ABI gets exposed?